Codecademy/gamut
# feat(Checkbox)!: require aria-label when label is ReactNode

### Overview


Updates `Checkbox` to require an aria label if the label is a ReactNode. Updates `ConnectedCheckbox` to take an aria label and pass it down to `Checkbox`. Also updates `Checkbox` border radius to be hard-coded at 4px instead of using `md` token so that the border radius is not impacted for other themes that have different border radii tokens (i.e. LX studio).


### PR Checklist

- [x] Related to JIRA ticket: [GM-855], [GM-1069]

#### Testing Instructions


1. Visit the Checkbox story
2. Confirm the border radius of the checkbox is 4px
3. Turn on VoiceOver for the Labels as ReactNodes section
4. Confirm "string" is read for the first checkbox (label used as aria-label), "a node" (aria-label since label is react node) is read for the second, and "here is a link to click" (aria-label since label is react node) is read for the third
5. Visit ConnectedForm story
6. Scroll down to Playground example
7. Turn on VoiceOver
8. Confirm "check it outtttt" label (label is <div>check it outtttt</div>) is read as "aria label" since that text is the aria label
9. To test that aria-label is required when label is a ReactNode, its easiest to do so locally
10. Go to packages/gamut/src/DataList/Controls/FilterControl.tsx
11. See that label is react node and aria label is provided and there are no TS errors
12. Comment out aria label and see you get a TS error that aria label is required
13. Change label to a string and see you do not get a TS error for not having aria label
14. Comment back in aria label and see you do not get a TS error for having aria label

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/39328)  | [Monolith Env](https://pr-39328-monolith.dev-eks.codecademy.com) |
| Mono       | [Mono Link](https://github.com/codecademy-engineering/mono/pull/9720)  | [Mono Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-9720)   |




[GM-855]: https://skillsoftdev.atlassian.net/browse/GM-855?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GM-1069]: https://skillsoftdev.atlassian.net/browse/GM-1069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
